### PR TITLE
Add Search API Documentation to docs.globus.org

### LINKED ISSUE #190: Begin plans for incorporating Search documentation
cc @ranantha, @teresas

We currently are hosting (beta) Search docs out of a bucket:
http://globus-search-docs.s3-website-us-east-1.amazonaws.com/stable/

It works, but presumably we want to bring these docs into docs.globus.org sometime soon.
@ranantha, when do we want to make that transition?

Note: I want to keep the docs rendering to that bucket for `master` so that we can easily preview doc updates there. That would mean keeping them possible to render under vanilla `asciidoc`.
@teresas, will that be possible? I assume that Transfer is doing this? I'll send you details on the source docs momentarily.

It would be nice to format consistently with the Transfer API docs, but I don't know what that would entail.

### PR DESCRIPTION
**Pull request by sirosen**
After a suitable amount of time spent wrangling nanoc into submission, I think I have a first version of the docs which can be posted on the docs site.

@jimPruyne, note that because the docs site provides its own menuing structures, and because link rendering is... different... there are actual content updates here. One of the big ones is that I just collapsed the whole `general/` section into the index page.
Also, I've removed the "previous schemas" section from the docs. I don't think we really need it, and it kind of explodes the menuing. If we wanted to add it back, which maybe we'd need one day, we could put it in but leave it out of the menus, and then link people there directly.

I want us to either (A) backport the docs into our repo to manage or (B) commit to just managing them purely here, in the docs site. The adaptations between how GitHub will want things, how raw asciidoctor will want things, and how nanoc (docs site) will want things could make this pretty painful. If we backport these changes, we may have to make further changes, but I think it's easier than trying to blindly produce content which works under nanoc or asking Teresa to do transformations of unknown/arbitrary complexity.

Could you please check this over, make sure there's nothing offensive to your sensibilities?

@teresas, I'm not expecting you to review all of the content, obviously, and I'm also okay with this going live with one or two broken links (though, obviously, I'd rather get it all right the first itme).
I've tried to check things over thoroughly, but maybe you can at least glance over the content and spot-check for "yeah, links on this page generally seem to render correctly and work".

Resolves #190

### CODE REVIEW

**Comment by jimPruyne** [comment]
TL;DR Looks ok by me without the benefit of a docs-site build, but I won't merge until docs owners agree with where it stands.

@sirosen @teresas I've read a bunch of the pages and I'm happy with how things look. As noted, there may be a broken link here or there, but short of compiling down to the actual docs site representation I'm doubtful I'd find them here regardless of my effort. I mostly viewed these with github's rendering as that made it easier to read, and perhaps would have caught some particularly bad formatting errors, but I didn't encounter any in my various read throughs (indeed github rendering was remarkably nice), and I don't think inter-page links are expected to work in this case anyway.

As to the question about how to maintain these going forward, I'd prefer not to backport. As long as the overall workflow is conducive to the Search team dropping in here and making changes and having those reviewed properly, then I'm most happy with this approach. We should do issues on the Search side to track those updates of course.

(Speaking of which, I do note that the overview doc states the use of HTTP 401 for missing auth consistent with recent actual code updates, but at least a couple pages I looked at for API calls didn't mention the 401 possibility as a return status. May want to do an issue on Search and attack here after merge happens).

**Comment by sirosen** [comment]
@teresas I've pushed changes which resolve these issues. Fingers crossed, it should all be good.
